dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -1025,11 +1025,26 @@ public void deleteProcessDefinitionByCode(User loginUser, long code) {
             logger.error("Delete process definition error, processDefinitionCode:{}.", code);
             throw new ServiceException(Status.DELETE_PROCESS_DEFINE_BY_CODE_ERROR);
         }
-        int deleteRelation = processTaskRelationMapper.deleteByCode(project.getCode(), processDefinition.getCode());
-        if (deleteRelation == 0) {
-            logger.warn(
+        processDefinitionLogMapper.deleteByProcessDefinitionCode(code);
+        List<ProcessTaskRelation> processTaskRelations =
+                processTaskRelationMapper.queryByProcessCode(project.getCode(), code);
+        if (!processTaskRelations.isEmpty()) {
+            int deleteRelation = processTaskRelationMapper.deleteByCode(project.getCode(), code);
+            if (deleteRelation == 0) {
+                logger.warn(
                     "The process definition has not relation, it will be delete successfully, processDefinitionCode:{}.",
                     code);
+            } else {
+                processTaskRelationLogMapper.deleteByProcessCode(code);
+                // delete relation task definition
+                List<Long> taskCodeList = processTaskRelations.stream().map(ProcessTaskRelation::getPostTaskCode)
+                        .collect(Collectors.toList());
+                int deleteTask = taskDefinitionMapper.deleteByCodeList(taskCodeList);
+                if (deleteTask == 0) {
+                    throw new ServiceException(Status.DELETE_TASK_DEFINE_BY_CODE_ERROR);
+                }
+                taskDefinitionLogMapper.deleteByCodeList(taskCodeList);
```

Comment:
task code needed to be deleted in `taskDefinitionLog` should get from `processTaskRelationLogMapper` by another query